app/src/main/java/org/mozilla/tv/firefox/tabs/WebRenderTabsTray.kt

```diff
@@ -0,0 +1,105 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.tabs
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import io.reactivex.functions.Consumer
+import mozilla.components.concept.tabstray.TabsTray
+import org.mozilla.tv.firefox.R
+import org.mozilla.tv.firefox.ext.webRenderComponents
+import org.mozilla.tv.firefox.pocket.PocketViewModel
+
+const val DEFAULT_ITEM_TEXT_COLOR = 0xFF111111.toInt()
+const val DEFAULT_ITEM_TEXT_SELECTED_COLOR = 0xFFFFFFFF.toInt()
+
+/**
+ * Tabs tray for NavigationOverlay
+ */
+class WebRenderTabsTray @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0,
+    private val tabsAdapter: TabsAdapter = TabsAdapter()
+) : RecyclerView(context, attrs, defStyleAttr),
+    TabsTray by tabsAdapter {
+
+    internal val styling: TabsTrayStyling
```

Comment:
Why do we need to style these programmatically instead of in the XML?  Could you add a comment explaining?